File: backend/src/main/java/com/example/contact/application/port/in/DeleteContactUseCase.java
package com.example.contact.application.port.in;

import com.example.contact.domain.exception.ContactNotFoundException;
import com.example.contact.domain.valueobject.ContactId;

/**
 * Inbound port for deleting a contact.
 */
public interface DeleteContactUseCase {

    /**
     * Deletes a contact by id.
     *
     * @param id contact id
     * @throws ContactNotFoundException when contact does not exist
     */
    void execute(ContactId id);
}

File: frontend/src/contact/application/port/in/DeleteContactUseCase.ts
import { ContactId } from '../../../domain/valueobject/ContactId';

/**
 * Inbound port for deleting a contact.
 */
export interface DeleteContactUseCase {
  /**
   * Deletes a contact by id.
   * @param id - contact id
   */
  execute(id: ContactId): Promise<void>;
}
